File: 0350.两个数组的交集II/0350-两个数组的交集II.go
func intersect(nums1 []int, nums2 []int) []int {
    if( len(nums1) > len(nums2) ) {
        return intersect(nums2, nums1)
    }

    m := map[int]int{}
    for _,num := range nums1 {
        m[num]++
    }

    ans := []int{}
    for _,num := range nums2 {
        if( m[num] > 0 ) {
            ans = append( ans, num);
            m[num]--
        }
    }

    return ans
}

File: 0350.两个数组的交集II/0350-两个数组的交集II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;

        map<int, int> dict;
        for( auto num : nums1 ) {
            dict[num]++;
        }        

        for( auto num : nums2 ) {
            if( dict.find(num) != dict.end() && dict[num] > 0 ) {
                dict[num]--;
                res.push_back(num);
            }
        }

        return res;
    }
};